tsdb/shard.go

```diff
@@ -121,18 +140,18 @@ func (s *Shard) Open() error {
 
 		// Open engine.
 		if err := s.engine.Open(); err != nil {
-			return fmt.Errorf("open engine: %s", err)
+			return err
 		}
 
 		// Load metadata index.
 		if err := s.engine.LoadMetadataIndex(s, s.index, s.measurementFields); err != nil {
-			return fmt.Errorf("load metadata index: %s", err)
+			return err
 		}
 
 		return nil
 	}(); err != nil {
 		s.close()
```

Comment:
I know this isn't directly related to your change, but this could easily be done with a named return parameter, and a deferred check at the beginning:
```
func (s *Shard) Open() (err error) {
    defer func() { if err != nil { s.close() } }()
    [...]
}
```

It would get rid of the large, multi-line anonymous function that is the true body of this function.